configure.ac

```diff
@@ -1061,20 +1061,10 @@ dnl             is a dummy function which no system supplies.  It is here to mak
 dnl             the system compile semi-correctly on OpenBSD which doesn't know
 dnl             how to create an empty archive
 dnl
-dnl     scandir: Only on BSD.
```

Comment:
Interestingly I don't see where we had a prototype for this one.

`_GNU_SOURCE` isn't listed in the `scandir(3)` man page for `scandir()` or `alphasort()`, but rather for `versionsort()` and `scandirat()`... I'd think/hope that `_GNU_SOURCE` is a sufficient condition for both of those though, for `glibc < 2.10`. If such an old `glibc` is even present in our supported distros.
```
   Feature Test Macro Requirements for glibc (see feature_test_macros(7)):

       scandir(), alphasort():
           /* Since glibc 2.10: */ _POSIX_C_SOURCE >= 200809L
               || /* Glibc <= 2.19: */ _BSD_SOURCE || _SVID_SOURCE

       versionsort():
           _GNU_SOURCE

       scandirat():
           _GNU_SOURCE
```

Edit: Found answer in `feature_test_macros(7)`:
```
In glibc versions before 2.20, defining _GNU_SOURCE also had the effect
              of implicitly defining _BSD_SOURCE and _SVID_SOURCE.
```

So we're probably fine since `_POSIX_C_SOURCE` covers it in newer versions.

Keeping this comment for posterity